File: Process.java
public class Process {
    private int burstTime;
    private String name;
    private int arriveTime;
    private boolean isAdded;

    public Process(String name,int burstTime,int arriveTime){
        this.name=name;
        this.burstTime=burstTime;
        this.arriveTime=arriveTime;
        this.isAdded=false;
    }

    public String getName(){ return this.name; }

    public int getBurstTime(){
        return this.burstTime;
    }

    public int getArriveTime(){
        return this.arriveTime;
    }

    public void addToTheHeap(){ this.isAdded=true;}

    public boolean isAddedToTheHeap(){
        return isAdded;
    }
}

File: Process.js
class Process {
    constructor(pid,name, arrivalTime, burstTime) {
        this.pid = pid;
        this.name = name;
        this.arrivalTime = arrivalTime;
        this.burstTime = burstTime;
        this.startTime=null;
        this.isAdded=false;
    }


    getArrivalTime(){
        return this.arrivalTime;
    }

    getBurstTime(){
        return this.burstTime;
    }

    isAddedToQueue(){
        return this.isAdded;
    }

    addToTheQueue(){
        this.isAdded=true;
    }


    getStartTime(){
        return this.startTime;
    }

    setStartTime(time){
        this.startTime=time;
    }

    getName(){
        return this.name;
    }

    getId123(){
        return 1;
    }

}